docs/content/any/reference/tooling/tracing.md

```diff
@@ -7,67 +7,36 @@ description: Polar tracing shows logs of how a query is evaluated.
+ 
+ * Query successes and the contents of any variable bindings returned by Oso.
+ 
+## Enabling `TRACE` output
 
-* Queries and sub-query evaluations.
+`TRACE` logs are viewed by setting an environment variable: `POLAR_LOG=trace`.
 
+`POLAR_LOG=trace` exposes a more verbose view of a query execution in Polar and extends the contents of `POLAR_LOG=info`. The `POLAR_LOG=trace` traces will emit a log for every Goal encountered by the Polar virtual machine during the query execution. Notable examples include:
 
-* Showing values that are bound to variables for each expression.
-
-
-* Showing which rules a Polar predicate call will evaluate.
+* Showing parameter type checking information during query applicability tests
 
+* Showing values that are bound to variables for each expression.
 
 * Showing values returned from calls into the application.
 
+
 ### Example
 
-```
-query> f(12);
-[debug]   QUERY: f(12), BINDINGS: {}
-[debug]     APPLICABLE_RULES: [
-[debug]       f(x) if new (Foo{x: x}, _instance_2) and .(_instance_2, foo(), _value_1) and _value_1 = x;
-[debug]       f(x) if x = 1;
-[debug]       f(x) if x + 1 and _op_3 == 2;
-[debug]     ]
-[debug]     RULE:
-[debug]     f(x) if
-[debug]       new Foo(x: x).foo() = x
-[debug]       QUERY: new (Foo{x: _x_10}, _instance_2_11) and .(_instance_2_11, foo(), _value_1_12) and _value_1_12 = _x_10, BINDINGS: {"_x_10": "12"}
-[debug]         QUERY: new (Foo{x: _x_10}, _instance_2_11), BINDINGS: {"_x_10": "12"}
-[debug]         QUERY: .(_instance_2_11, foo(), _value_1_12) and _value_1_12 = _x_10, BINDINGS: {"_instance_2_11": "Foo{x: 12}", "_x_10": "12"}
-[debug]           QUERY: .(_instance_2_11, foo(), _value_1_12), BINDINGS: {"_instance_2_11": "Foo{x: 12}"}
-[debug]             LOOKUP: Foo{x: 12}.foo()
-[debug]             => 13
-[debug]           QUERY: _value_1_12 = _x_10, BINDINGS: {"_x_10": "12", "_value_1_12": "13"}
-[debug]             BACKTRACK
-[debug]             LOOKUP: Foo{x: 12}.foo()
-[debug]             => No more results.
-[debug]             BACKTRACK
-[debug]     RULE:
-[debug]     f(x) if
-[debug]       x = 1
-[debug]       QUERY: _x_13 = 1, BINDINGS: {"_x_13": "12"}
-[debug]         BACKTRACK
-[debug]     RULE:
-[debug]     f(x) if
-[debug]       x + 1 == 2
-[debug]       QUERY: _x_14 + 1 and _op_3_15 == 2, BINDINGS: {"_x_14": "12"}
-[debug]         QUERY: _x_14 + 1, BINDINGS: {"_x_14": "12"}
-[debug]           MATH: 12 + 1 = _op_3_15, BINDINGS: {}
-[debug]         QUERY: _op_3_15 == 2, BINDINGS: {"_op_3_15": "13"}
-[debug]           CMP: 13 == 2, BINDINGS: {}
-[debug]           BACKTRACK
-[debug]           HALT
-False
-```
+{{<literalInclude path="reference/tooling/info" tabGroup="tracing">}}
+{{<literalInclude path="reference/tooling/trace" tabGroup="tracing">}}
 
-Debug mode can be disabled by setting `POLAR_LOG=0` or `POLAR_LOG=off`.
\ No newline at end of file
+All trace output can be disabled by setting `POLAR_LOG=0` or `POLAR_LOG=off`.
```

Comment:
to make it clear that these will disable all info/debug output in addition to all trace output
```suggestion
All `POLAR_LOG` output can be disabled by setting `POLAR_LOG=0` or `POLAR_LOG=off`.
```